test/framework/src/main/java/org/elasticsearch/transport/MockTcpTransport.java

```diff
@@ -261,20 +261,50 @@ public long serverOpen() {
         private final CancellableThreads cancellableThreads = new CancellableThreads();
         private final Closeable onClose;
 
+        /**
+         * Constructs a new MockChannel instance intended as client.
```

Comment:
not sure exactly what you mean with "client". This is used both when this node connects to another node and when this nodes receives a connection from another one.